src/main/java/SasDefinitionSample.java

```diff
@@ -1,230 +1,215 @@
+        storageAccount = azureResourceManager
+                .storageAccounts().define(STORAGE_ACCOUNT_NAME)
+                .withRegion(VAULT_REGION)
+                .withExistingResourceGroup(RESOURCE_GROUP)
+                .withFileEncryption()
+                .withBlobEncryption()
+                .withGeneralPurposeAccountKindV2()
+                .withSystemAssignedManagedServiceIdentity()
+                .create();
+
+        // The key vault service must be given the "Storage Account Key Operator Service Role" on
+        // the storage account before the storage account can be added to the vault.
+        System.out.println("Granting Azure Key Vault the operator service role on the storage account");
+
+        // find the role definition for "Storage Account Key Operator Service Role"
+        Optional<RoleDefinition> operationDefinition =
+                azureResourceManager.accessManagement()
+                        .roleDefinitions().listByScope("/")
+                        .stream()
+                        .filter(roleDefinition -> "Storage Account Key Operator Service Role".equals(roleDefinition.roleName()))
+                        .findFirst();
+        if (operationDefinition.isPresent()) {
+            azureResourceManager.accessManagement().roleAssignments()
+                    .define(UUID.randomUUID().toString()) //Needs to be a UUID formatted String
+                    .forObjectId("93c27d83-f79b-4cb2-8dd4-4aa716542e74") //This is the Azure Key Vault Service Principal
+                    .withRoleDefinition(operationDefinition.get().id())
+                    .withScope(storageAccount.id())
+                    .create();
+        }
+
+        // Setting the storage acocunt can only be called by a user account with access to the keys of the storage account.
+        // Therefore, we grant the user that created the storage account access to the vault
+        // as well as the storage account.
+
+        System.out.println("Creating new vault");
+        vault = azureResourceManager
+                .vaults().define(VAULT_NAME)
+                .withRegion(VAULT_REGION)
+                .withExistingResourceGroup(RESOURCE_GROUP)
+                .defineAccessPolicy()
+                .forObjectId(USER_OID)
+                .allowSecretAllPermissions()
+                .allowStorageAllPermissions()
+                .allowCertificateAllPermissions()
+                .allowKeyAllPermissions()
+                .attach()
+                .defineAccessPolicy()
+                .forObjectId(storageAccount.innerModel().identity().principalId())
+                .allowKeyPermissions(KeyPermissions.UNWRAP_KEY, KeyPermissions.WRAP_KEY, KeyPermissions.GET)
+                .attach()
+                .withPurgeProtectionEnabled()
+                .withSoftDeleteEnabled()
+                .create();
+        VAULT_URI = vault.vaultUri();
+        Key keyVaultKey = vault.keys()
```

Comment:
See my previous comment about using a management plane library to create a key.

Reply:
Yes,  this is using the management plane library to create a key instead of the data plane library.